src/plugins/intel_gpu/src/kernel_selector/kernels/eltwise/eltwise_kernel_blocked_opt.cpp

```diff
@@ -0,0 +1,378 @@
+
+    DispatchData dispatchData = SetDefault(newParams);
+
+    auto& kernel = kd.kernels[0];
+
+    kernel.params.workGroups.global = dispatchData.gws;
+    kernel.params.workGroups.local = dispatchData.lws;
+
+    kernel.code.kernelString = GetKernelString(kernelName, jit, entry_point, params.engineInfo, EXE_MODE_DEFAULT);
+    kernel.params.arguments = GetArgsDesc((uint32_t)newParams.inputs.size(),
+                                   false,
+                                   false,
+                                   GetFusedPrimitiveInputsCount(params));
+
+    return {kd};
+}
+
+KernelsPriority EltwiseKernel_blocked_opt::GetKernelsPriority(const Params& /*params*/, const optional_params& /*options*/) const {
+    return FORCE_PRIORITY_1;
+}
+
+// Protected
+bool EltwiseKernel_blocked_opt::Validate(const Params& params, const optional_params& o) const {
+    if (!EltwiseKernelBase::Validate(params, o)) {
+        return false;
+    }
+
+    const auto& ewParams = static_cast<const eltwise_params&>(params);
+    if (IsUnsupportedModeForVecCode(ewParams))
+        return false;
+
+    for (size_t i = 0; i < ewParams.inputs.size(); i++) {
+        if ((GetFeatureBlockSizeFromFormat(ewParams, i) == 1) &&
+            !IsBroadcastingPossibleInput(ewParams.inputs[i], ewParams.outputs[0])) {
+            return false;
+        }
+    }
+
+    auto input0 = ewParams.inputs[0];
+    const auto vec_size = GetFeatureBlockSizeFromFormat(ewParams, 0);
+    const auto& output = ewParams.outputs[0];
+    // Check that padding before features doesn't miss-align the blocks
+    if (input0.Feature().pad.before % vec_size != 0 || output.Feature().pad.before % vec_size != 0) {
+        return false;
+    }
+
+    auto compareTensors = [](const DataTensor& input0, const DataTensor& input1) -> bool {
+        // Check all parameters except DataType
+        auto& input0_dims = input0.GetDims();
+        auto& input1_dims = input1.GetDims();
+        bool same = input0.GetLayout() == input1.GetLayout() &&
+                    input0.GetPaddedVal() == input1.GetPaddedVal() &&
+                    input0.GetViewOffset() == input1.GetViewOffset() &&
+                    input0_dims.size() == input1_dims.size();
+        if (same) {
+            for (size_t i = 0; i < input0_dims.size(); i++) {
+                same &= input0_dims[i].v == input1_dims[i].v &&
+                        input0_dims[i].pad.before == input1_dims[i].pad.before &&
+                        input0_dims[i].pad.after == input1_dims[i].pad.after &&
+                        input0_dims[i].pitch == input1_dims[i].pitch;
+            }
+        }
+        return same;
+    };
+
+    for (size_t i = 1; i < ewParams.inputs.size(); i++) {
+        if (ewParams.inputs[i].LogicalSize() == input0.LogicalSize() && !(compareTensors(ewParams.inputs[i], input0)))
+            return false;
+        if (ewParams.inputs[i].Feature().pad.before % vec_size != 0) {
```

Comment:
I think this is duplicated with line 105.

Reply:
This also affects rop_test/crop_gpu.pad_test